### DISCUSSION TITLE
Create Trigger in database directly?

### DISCUSSION THREAD

**Discussion by osseonews** [pocketbase/pocketbase#573] [Q&A]
I needed to create some triggers in the database. As I don't know how to program in Go and so can't use the Extend feature of pocketbase right now, my idea was simply to add those triggers on the database level directly using Sqlite database tools. I would then upload the new database with triggers to the pocketbase folder, and I assume everything would just work. Is there anything wrong with this approach?

**Comment by ganigeorgiev** [comment]
It depends on what the triggers do, but if they are used mostly for data integrity checks, validations, etc. it should work just fine.

If your application is using the realtime capabilities of PocketBase, you may stumble on some issues when updating a record with a plain db trigger since we currently stream record changes only if they were persisted with the `Dao` service.

**Comment by osseonews** [comment]
Thanks. Yes, I was going to use a trigger to update an inventory value in the database, as we are using the database to track inventory as orders come in. In theory, inventory data is great for SSE, but at this point its really only an internal application, so we can just query the server again to get update data. I guess I will do this for now, until I learn a bit of Go. Do you think it will be a problem if later on, I just drop the trigger in the database and recreate with the Dao service to take advantage of realtime?

**Reply by ganigeorgiev** [reply]
> Do you think it will be a problem if later on, I just drop the trigger in the database and recreate with the Dao service to take advantage of realtime?

No. In fact, since the `Dao` service operates on the application level, it is far more flexible and you may find it easier to work with then using the plain sql triggers. The documentation is currently lacking examples and details (we have only [godocs](https://pkg.go.dev/github.com/pocketbase/pocketbase/daos#pkg-types)), but that will be fixed in the near future.
